lib/internal/util/comparisons.js

```diff
@@ -5,30 +5,53 @@ const { isArrayBufferView } = require('internal/util/types');
 const { internalBinding } = require('internal/bootstrap/loaders');
 const { isDate, isMap, isRegExp, isSet } = internalBinding('types');
 
-function objectToString(o) {
-  return Object.prototype.toString.call(o);
+const ReflectApply = Reflect.apply;
+
+function uncurryThis(func) {
+  return (thisArg, ...args) => ReflectApply(func, thisArg, args);
 }
 
+const kStrict = true;
+const kLoose = false;
+
+const kNoIterator = 0;
+const kIsArray = 1;
+const kIsSet = 2;
+const kIsMap = 3;
+
+const objectToString = uncurryThis(Object.prototype.toString);
+const hasOwnProperty = uncurryThis(Object.prototype.hasOwnProperty);
+const getTime = uncurryThis(Date.prototype.getTime);
+const propertyIsEnumerable = uncurryThis(Object.prototype.propertyIsEnumerable);
+
+const objectKeys = Object.keys;
+const getPrototypeOf = Object.getPrototypeOf;
+const getOwnPropertySymbols = Object.getOwnPropertySymbols;
+const objectIs = Object.is;
+const numberIsNaN = Number.isNaN;
+
 // Check if they have the same source and flags
 function areSimilarRegExps(a, b) {
   return a.source === b.source && a.flags === b.flags;
 }
 
-// For small buffers it's faster to compare the buffer in a loop. The c++
-// barrier including the Uint8Array operation takes the advantage of the faster
-// binary compare otherwise. The break even point was at about 300 characters.
-function areSimilarTypedArrays(a, b, max) {
+function areSimilarFloatArrays(a, b) {
   const len = a.byteLength;
```

Comment:
Is the caching of `len` still helpful/needed?

Reply:
No

Reply:
Then can you remove it :)?